File: java/Stack/easy/crawler-log-folder.java
class Solution {
    public int minOperations(String[] logs) {
        int c=0;
        for(String str: logs){
            if (str.charAt(1)=='/'){c+=0;}
            else if(str.charAt(1)=='.'){
                if(c==0){c+=0;}
                else{c--;}
            }
            else{
                c++;
            }
        }
    return c;
    }
}

File: cpp/Stack/easy/crawler-log-folder.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int curr = 0;
        for(auto i : logs) {
            if(i == "./") {
                // do nothing
            } else if(i[0] == '.') {
                curr--;
            } else {
                curr++;
            }
            curr = max(curr, 0);
        }
        return curr;
    }
};
